src/build.d

```diff
@@ -0,0 +1,1131 @@
+    {
+        t = t.buildNormalizedPath; // remove trailing slashes
+        switch (t)
+        {
+            case "auto-tester-build":
+                "TODO: auto-tester-all".writeln; // TODO
+                break;
+
+            case "toolchain-info":
+                "TODO: info".writeln; // TODO
+                break;
+
+            case "unittest":
+                "TODO: unittest".writeln; // TODO
+                break;
+
+            case "cxx-unittest":
+                "TODO: cxx-unittest".writeln; // TODO
+                break;
+
+            case "check-examples":
+                "TODO: cxx-unittest".writeln; // TODO
+                break;
+
+            case "build-examples":
+                "TODO: build-examples".writeln; // TODO
+                break;
+
+            case "checkwhitespace":
+                "TODO: checkwhitespace".writeln; // TODO
+                break;
+
+            case "html":
+                "TODO: html".writeln; // TODO
+                break;
+
+            case "install":
+                "TODO: install".writeln; // TODO
+                break;
+
+            case "man":
+                "TODO: man".writeln; // TODO
+                break;
+
+            dmd:
+            case "dmd":
+                buildDMD();
+                break;
+
+            case "clean":
+                env["G"].rmdirRecurse;
+                exit(0);
+                break;
+
+            default:
+            case "all":
+                goto dmd;
+                break;
+        }
+    }
+    return newTargets.data;
+}
+
+// Sets the environment variables
+string[string] getEnvironment()
+{
+    string[string] env;
+
+    env.getDefault("TARGET_CPU", "X86");
+    auto os = env.getDefault("OS", detectOS);
+    auto build = env.getDefault("BUILD", "release");
+    enforce(build.among("release", "debug"), "BUILD must be 'debug' or 'release'");
+
+    // detect Model
+    auto model = env.getDefault("MODEL", detectModel);
+    env["MODEL_FLAG"] = "-m" ~ env["MODEL"];
+
+    // detect PIC
+    version(Posix)
+    {
+        // default to PIC on x86_64, use PIC=1/0 to en-/disable PIC.
+        // Note that shared libraries and C files are always compiled with PIC.
+        bool pic;
+        version(X86_64)
+            pic = true;
+        else version(X86)
+            pic = false;
+        if (environment.get("PIC", "0") == "1")
+            pic = true;
+
+        env["PIC_FLAG"]  = pic ? "-fPIC" : "";
+    }
+    else
+    {
+        env["PIC_FLAG"] = "";
+    }
+
+    env.getDefault("GIT_HOME", "https://github.com/dlang");
+    env.getDefault("SYSCONFDIR", "/etc");
+    env.getDefault("TMP", "/tmp");
+    env.getDefault("PGO_DIR", srcDir.buildPath("pgo"));
+    auto d = env.getDefault("D", "dmd");
+    env.getDefault("C", d.buildPath("backend"));
+    env.getDefault("TK", d.buildPath("tk"));
+    env.getDefault("ROOT", d.buildPath("root"));
+    env.getDefault("EX", d.buildPath("examples"));
```

Comment:
This variable isn't used yet, but it will very likely soon be needed (i.e. when the example goal gets added)